R/compute_duration.R

```diff
@@ -151,19 +151,35 @@ compute_duration <- function(start_date,
                              add_one = TRUE,
                              trunc_out = FALSE,
                              type = "duration") {
+  mapping <- c(
+    years = c("year", "years", "yr", "yrs", "y"),
+    months = c("month", "months", "mo", "mos"),
+    weeks = c("week", "weeks", "wk", "wks", "w"),
+    days = c("day", "days", "d"),
+    hours = c("hour", "hours", "hr", "hrs", "h"),
+    minutes = c("minute", "minutes", "min", "mins"),
+    seconds = c("second", "seconds", "sec", "secs", "s")
+  )
+  in_unit <- names(mapping[mapping %in% tolower(in_unit)]) %>% str_replace(., '[0-9]', '')
+  out_unit <- names(mapping[mapping %in% tolower(out_unit)]) %>% str_replace(., '[0-9]', '')
```

Comment:
@bms63 @bundfussr  @manciniedoardo Take a look here at lines 154-164, which make it easily modifiable what we accept/want to map onto our duration calculations, see corresponding tests in `test-compute_duration.R`

Reply:
Ahh yes totally forgot that was introduced, we would have to upversion of description files for dplyr (>=1.1.0) for that though

Reply:
> Ahh yes totally forgot that was introduced, we would have to upversion of description files for dplyr (>=1.1.0) for that though

If we do not want to upversion, we could use `case_match()`. @bms63 , @manciniedoardo , what do you think?

Reply:
I'm hesitant to fix to 1.1.0 as it has only been out a year.  Maybe simpler isn't the solution here.

Reply:
Well both solutions are already drafted, only a matter of deciding which to proceed with, for easier review purposes I've linked the commit associated with both below:

1. no `case_match` from [this commit](https://github.com/pharmaverse/admiral/pull/2261/commits/d6c550eee01665b9430260381cdadf1639b0497c)

2. with `case_match`  where it stands [right now](https://github.com/pharmaverse/admiral/pull/2261/files)

Reply:
Given the nature of Pharma I would be want to pick option 1.

Reply:
> If we do not want to upversion, we could use `case_match()`.

Sorry, I meant `case_when()`. I thinks even with `case_when()` the code is much easier to read.

Furthermore, as we need this mapping more than once I would create a (non-exported) function for it.